### DISCUSSION TITLE
Artifacts when using programmatically generated SVG string

### DISCUSSION THREAD

**Discussion by sanbai0** [visgl/deck.gl#6790] [Q&A] [answered]
A puzzling mistake has occurred when I use image from a programmatically generated SVG string.
I use `new IconLayer({
    id: 'icon-layer',
    getIcon: d => ({
      url: d.dataUrl,
      width: d.width,
      height: d.height,
    })`
But IconLayer Can't draw SVG image correctly.

The SVG image What I want
![image](https://user-images.githubusercontent.com/37434602/161018314-c8c68752-9bce-4235-8b50-770d9c431068.png)
IconLayer drawn incorrectly：
![image](https://user-images.githubusercontent.com/37434602/161017487-203b3b9a-5e62-4fc0-adba-f1d3b59f9ae4.png)

SVG code：
`<svg width="147" height="50">
<polygon stroke-width="1.2" stroke="#506882" fill="#ffffff" points="125.3681640625,16,12,16,4,24,12,33,125.3681640625,33,125.3681640625,16"></polygon>
<polygon stroke-width="1.2" stroke="#506882" fill="#506882" points="124.3681640625,16,146,16,146,33,124.3681640625,33"></polygon>
<text stroke-width="2" fill="#506882" x="67.68408203125" y="9.5" style="text-anchor: middle; dominant-baseline: middle; font: 11px sans-serif;">159.2</text>
<text stroke-width="2" fill="#506882" x="67.68408203125" y="25" style="text-anchor: middle; dominant-baseline: middle; font: 11px sans-serif;">ESC07/ESC08/ESC09</text>
<text stroke-width="2" fill="#506882" x="67.68408203125" y="42" style="text-anchor: middle; dominant-baseline: middle; font: 11px sans-serif;">//</text>
<text stroke-width="2" fill="#ffffff" x="134.18408203125" y="25" style="text-anchor: middle; dominant-baseline: middle; font: 11px sans-serif;">R4</text>
</svg>`

dataUrl from SVG string：
[data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22147%22%20height%3D%2250%22%3E%3Cpolygon%20stroke-width%3D%221.2%22%20stroke%3D%22%23506882%22%20fill%3D%22%23ffffff%22%20points%3D%22125.3681640625%2C16%2C12%2C16%2C4%2C24%2C12%2C33%2C125.3681640625%2C33%2C125.3681640625%2C16%22%2F%3E%3Cpolygon%20stroke-width%3D%221.2%22%20stroke%3D%22%23506882%22%20fill%3D%22%23506882%22%20points%3D%22124.3681640625%2C16%2C146%2C16%2C146%2C33%2C124.3681640625%2C33%22%2F%3E%3Ctext%20stroke-width%3D%222%22%20fill%3D%22%23506882%22%20x%3D%2267.68408203125%22%20y%3D%229.5%22%20style%3D%22text-anchor%3A%20middle%3B%20dominant-baseline%3A%20middle%3B%20font%3A%2011px%20sans-serif%3B%22%3E159.2%3C%2Ftext%3E%3Ctext%20stroke-width%3D%222%22%20fill%3D%22%23506882%22%20x%3D%2267.68408203125%22%20y%3D%2225%22%20style%3D%22text-anchor%3A%20middle%3B%20dominant-baseline%3A%20middle%3B%20font%3A%2011px%20sans-serif%3B%22%3EESC07%2FESC08%2FESC09%3C%2Ftext%3E%3Ctext%20stroke-width%3D%222%22%20fill%3D%22%23506882%22%20x%3D%2267.68408203125%22%20y%3D%2242%22%20style%3D%22text-anchor%3A%20middle%3B%20dominant-baseline%3A%20middle%3B%20font%3A%2011px%20sans-serif%3B%22%3E%2F%2F%3C%2Ftext%3E%3Ctext%20stroke-width%3D%222%22%20fill%3D%22%23ffffff%22%20x%3D%22134.18408203125%22%20y%3D%2225%22%20style%3D%22text-anchor%3A%20middle%3B%20dominant-baseline%3A%20middle%3B%20font%3A%2011px%20sans-serif%3B%22%3ER4%3C%2Ftext%3E%3C%2Fsvg%3E](url)

How can I get correct SVG image？

**Comment by Pessimistress** [comment]
You are missing `xmlns="http://www.w3.org/2000/svg"` on the SVG. A simple way to test your data URL is to copy and paste it into the address bar of your browser. It should display the image if the URL is constructed correctly.

See https://codepen.io/Pessimistress/pen/VwyyGwd?editors=0010

**Reply by sanbai0** [reply]
As you can see，It's still a wrong image.
<img width="1024" alt="7C970DE3-2064-455a-B3A2-EB22E755C7E4" src="https://user-images.githubusercontent.com/37434602/161877925-b0af7dc0-ac70-4546-b8ba-e2998ff36aff.png">

**Reply by Pessimistress** [reply]
Works fine for me. Please share your OS/browser.

**Reply by sanbai0** [reply]
windows10

Microsoft Edge
Version 99.0.1150.55 (official version) (64 bit)

**Comment by Pessimistress** [comment, marked as answer]
I can confirm that calling `createImageBitmap` with a SVG image in Windows Edge and Chrome generates these artifacts around text elements. FF on Windows works as expected. @ibgreen 

Repro: https://codepen.io/Pessimistress/pen/rNpdgBw

@sanbai0 You should report this as a bug to https://bugs.chromium.org/p/chromium/issues/list

You can work around this by adding the following prop to the IconLayer:
```js
loadOptions: {
  image: {
    type: 'image'
  }
}
```
